app/js/lib/foxbox/services/door-lock.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+import BaseService from './base';
+
+const TYPE = 'door-lock';
+
+export default class DoorLockService extends BaseService {
+  constructor(props, config) {
+    super(props, config);
+    Object.seal(this);
+  }
+
+  get type() {
+    return TYPE;
+  }
+
+  /**
+   * Checks if the lock is locked.
+   *
+   * @returns {Promise.<boolean>}
+   */
+  isLocked() {
+    return this.get('DoorLocked')
+      .then((response) => {
+        if (!response) {
+          throw new Error('Door lock status is not available yet!');
+        }
+
+        return response.DoorLocked === 'Locked';
+      });
+  }
+
+  /**
+   * Toggles state of the lock depending on the "isLocked" parameter.
+   *
+   * @param isLocked Boolean value indicating whether we want lock or unlock
+   * the lock.
+   * @returns {Promise}
+   */
+  toggle(isLocked) {
```

Comment:
I find `toggle()` confusing because the common case for a function called `toggle` is to reverse the state without passing parameters (e.g. `el.classList.toggle('highlighted');`).
But even `lock()` sounds confusing.